src/Integrations/Greenhouse/GreenhouseClient.php

```diff
@@ -135,15 +140,28 @@ public function postApplication(string $itemId, array $params, array $files, str
 			$this->prepareFiles($files)
 		);
 
-		$response = \wp_remote_post(
-			self::BASE_URL . "boards/{$this->getBoardToken()}/jobs/{$itemId}",
+		$filterName = Filters::getGeneralSettingsFilterName('httpRequestTimeout');
+
+		// Curl used because files are not sent via wp request.
+		$curl = \curl_init(); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_init
+		\curl_setopt_array( // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_setopt_array
+			$curl,
 			[
-				'headers' => $this->getHeaders(true),
-				'body' => \wp_json_encode($body),
+				\CURLOPT_URL => self::BASE_URL . "boards/{$this->getBoardToken()}/jobs/{$itemId}",
+				\CURLOPT_HTTPAUTH => \CURLAUTH_BASIC,
+				\CURLOPT_RETURNTRANSFER => true,
+				\CURLOPT_TIMEOUT => \apply_filters($filterName, General::HTTP_REQUEST_TIMEOUT_DEFAULT),
+				\CURLOPT_POST => true,
+				\CURLOPT_POSTFIELDS => $body,
+				\CURLOPT_HTTPHEADER => $this->getHeaders(true),
 			]
 		);
+		$response = \curl_exec($curl); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_exec
+		$code = \curl_getinfo($curl, \CURLINFO_RESPONSE_CODE); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_getinfo
```

Comment:
How come you are using curl instead of `wp_remote_request`? Remote request is a wrapper around curl/wget iirc, and should have the same parameters.

Reply:
Hmm, odd. Did you manage to see what is changed/modified when using `wp_remote_request` vs `curl_exec`?

It's a wrapper around the [requests lib](https://github.com/WordPress/Requests/) which is pretty powerful.